pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -131,6 +133,70 @@ func TestShouldIgnoreComm(t *testing.T) {
 	}
 }
 
+func TestLoadIgnoredComm(t *testing.T) {
+	tests := []struct {
+		name   string
+		comms  string
+		expect []bool
+	}{
+		{
+			name:   "empty list of ignored commands",
+			comms:  "",
+			expect: []bool{false},
+		},
+		{
+			name:   "short commands in config list",
+			comms:  "cron, polkitd, rsyslogd, bash, sshd, dockerd",
+			expect: []bool{true, true, true, true, true, true},
+		},
+		{
+			name:   "malformed commands list",
+			comms:  "rsyslogd, , snapd, ,   udisksd, containerd, ,    ",
+			expect: []bool{true, false, true, false, true, true, false, false},
+		},
+		{
+			name:   "long commands in config list",
+			comms:  "containerd-shim-runc-v2,unattended-upgrade-shutdown,kube-controller-manager",
+			expect: []bool{true, true, true},
+		},
+		{
+			name:   "commands of different lengths in the configuration list",
+			comms:  "containerd-shim-runc-v2,calico-node,unattended-upgrade-shutdown,bash,kube-controller-manager",
+			expect: []bool{true, true, true, true, true},
+		},
+	}
+	defLen := len(ignoreComms)
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockSystemProbe := mock.NewSystemProbe(t)
+			require.NotEmpty(t, mockSystemProbe)
+
+			mockSystemProbe.SetWithoutSource("discovery.ignore_comms", test.comms)
+
+			LoadIgnoredComms(newConfig())
+
+			comms := strings.Split(strings.ReplaceAll(test.comms, " ", ""), ",")
+			if len(comms) > 0 {
+				if len(comms[0]) == 0 {
+					require.Equal(t, len(ignoreComms), defLen)
+				} else {
+					require.Greater(t, len(ignoreComms), defLen)
+				}
+			}
```

Comment:
That check is meaningless
for instance, try to move the first scenario to be the second scenario, the test will fail.
Also, don't check `Greater`. You know exactly how ignoreComms should look like

Reply:
I've refactored this test case